internal/impl/deltalake/output.go

```diff
@@ -0,0 +1,202 @@
+package deltalake
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/redpanda-data/benthos/v4/public/service"
+	"github.com/rivian/delta-go"
+	"github.com/rivian/delta-go/lock/nillock"
+	"github.com/rivian/delta-go/state"
+	"github.com/rivian/delta-go/storage"
+	"github.com/rivian/delta-go/storage/s3store"
+)
+
+const (
+	dlFieldTableURL = "table_url"
+	dlFieldBatching = "batching"
```

Comment:
We should probably make this a batching output

Reply:
It's not used, do we need it?

Reply:
@rockwotj Ok, I wasn't sure about this because with parquet the messages are already sort of in a batch. Do you think it's still useful to use a batching output in that case?